File: mycollab-web-pro/src/main/java/com/mycollab/pro/module/file/view/FileController.kt
package com.mycollab.pro.module.file.view

import com.google.common.eventbus.Subscribe
import com.mycollab.eventmanager.ApplicationEventListener
import com.mycollab.module.file.event.FileEvent
import com.mycollab.vaadin.mvp.AbstractController
import com.mycollab.vaadin.mvp.PresenterResolver

/**
 * @author MyCollab Ltd
 * @since 6.0.0
 */
class FileController(container: FileModule) : AbstractController() {
    init {
        this.register(object : ApplicationEventListener<FileEvent.GotoList> {
            @Subscribe
            override fun handle(event: FileEvent.GotoList) {
                val presenter = PresenterResolver.getPresenter(FileMainPresenter::class.java)
                presenter.go(container, null)
            }
        })
    }
}

File: mycollab-web-pro/src/main/scala/com/mycollab/pro/module/file/view/FileController.scala
package com.mycollab.pro.module.file.view

import com.mycollab.eventmanager.ApplicationEventListener
import com.mycollab.vaadin.mvp.{AbstractController, PresenterResolver}
import com.google.common.eventbus.Subscribe
import com.mycollab.module.file.event.FileEvent

/**
  * @author MyCollab Ltd
  * @since 5.0.9
  */
class FileController(container: FileModule) extends AbstractController {
  bindFileEvents()

  private def bindFileEvents() {
    this.register(new ApplicationEventListener[FileEvent.GotoList]() {
      @Subscribe def handle(event: FileEvent.GotoList) {
        val presenter = PresenterResolver.getPresenter(classOf[FileMainPresenter])
        presenter.go(container, null)
      }
    })
  }
}
